seatunnel-apis/seatunnel-api-spark/src/main/scala/org/apache/seatunnel/spark/BaseSparkSink.java

```diff
@@ -14,19 +14,28 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.seatunnel.spark
 
-import org.apache.seatunnel.apis.BaseSource
-import org.apache.seatunnel.shade.com.typesafe.config.{Config, ConfigFactory}
+package org.apache.seatunnel.spark;
 
-trait BaseSparkSource[Data] extends BaseSource[SparkEnvironment] {
+import org.apache.seatunnel.apis.BaseSink;
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigFactory;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
 
-  protected var config: Config = ConfigFactory.empty()
+public abstract class BaseSparkSink<OUT>  implements BaseSink<SparkEnvironment> {
+    protected Config config = ConfigFactory.empty();
 
-  override def setConfig(config: Config): Unit = this.config = config
+    @Override
+    public void setConfig(Config config){
+        this.config = config;
+    }
 
-  override def getConfig: Config = config
+    @Override
+    public Config getConfig(){
+        return config;
+    }
 
-  def getData(env: SparkEnvironment): Data;
+    public abstract OUT output(Dataset<Row> data, SparkEnvironment env);
```

Comment:
Why modify input parameters and output types?

Reply:
the origin code is below.
```
  def output(data: Dataset[Row], env: SparkEnvironment): OUT;

```
Could you describe more detial?

Reply:
> the origin code is below.
> 
> ```
>   def output(data: Dataset[Row], env: SparkEnvironment): OUT;
> ```
> 
> Could you describe more detial?

Am I reading it wrong?

Reply:
Git Record make confuse.
The real  code different is below.
![image](https://user-images.githubusercontent.com/18548053/150716806-90186fff-c15d-48fc-aa23-54b952e0c9cf.png)